waspc/src/Wasp/Analyzer/Evaluator/EvaluationError.hs

```diff
@@ -21,7 +21,11 @@ data EvaluationError
   | -- | "MissingField fieldName"
     MissingField String
   | -- | In case when evaluation includes parsing with Parsec and it fails.
-    ParseError ParseError
+    ParseErrorParsec ParseError
+  | -- | In case when evaluation includes parsing and it fails.
+    -- TODO: This is a very broad error, to help us capture custom parsing errors.
+    --   Consider if it is too broad and if we should make it more granular / precise.
+    ParseError String
```

Comment:
Refactor this into `ParseError ParseError` and then `data ParseError = ParseErrorParsec Parsec.ParseError | ParseErrorString String`?